src/main/java/net/glowstone/entity/passive/GlowWolf.java

```diff
@@ -1,39 +1,43 @@
 package net.glowstone.entity.passive;
 
+import com.flowpowered.networking.Message;
+import net.glowstone.entity.meta.MetadataIndex;
+import net.glowstone.entity.meta.MetadataIndex.TameableFlags;
 import org.bukkit.DyeColor;
 import org.bukkit.Location;
 import org.bukkit.entity.EntityType;
 import org.bukkit.entity.Wolf;
 
-//TODO: Further implementation
+import java.util.List;
+import java.util.Random;
+
 public class GlowWolf extends GlowTameable implements Wolf {
 
-    private boolean angry;
-    private boolean sitting;
     private DyeColor collarColor;
 
     public GlowWolf(Location location) {
-        super(location, EntityType.WOLF);
-    }
-
-    @Override
-    public boolean isAngry() {
-        return angry;
+        super(location, EntityType.WOLF, 8);
+        Random r = new Random();
+        collarColor = DyeColor.getByData((byte) r.nextInt(DyeColor.values().length));
     }
 
     @Override
-    public void setAngry(boolean angry) {
-        this.angry = angry;
+    public List<Message> createUpdateMessage() {
+        if (isTamed()) {
+            metadata.set(MetadataIndex.WOLF_OWNER, getOwner().getName());
+        }
+        metadata.set(MetadataIndex.WOLF_HEALTH, (float) getHealth());
+        return super.createUpdateMessage();
     }
 
     @Override
-    public boolean isSitting() {
-        return sitting;
+    public boolean isAngry() {
+        return metadata.getBit(MetadataIndex.WOLF_FLAGS, TameableFlags.WOLF_IS_ANGRY);
     }
 
     @Override
-    public void setSitting(boolean sitting) {
-        this.sitting = sitting;
+    public void setAngry(boolean angry) {
+        metadata.setBit(MetadataIndex.WOLF_FLAGS, TameableFlags.WOLF_IS_ANGRY, angry);
     }
```

Comment:
Same thing as in Ocelot: (setSitting() and isSitting())
For API (Glowkit) Support call super methods here?